### PR TITLE
Homepage without database hits

### PR DESCRIPTION
**Pull request by yakloinsteak**
## Description

We think that slow probe times are due to database load, and the probes hit the home page. 

I'm not sure how frequently the links table changes, so we could use redis or some other preferred way of memoizing in this application.

## Type of change
- [x] Code clean-up / housekeeping

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

**Comment by yakloinsteak** [comment]
@schwartzkylej fyi.

File: app/models/link.rb [hunk 1]
```diff
@@ -5,8 +5,10 @@
 ###
 
 class Link < ApplicationRecord
-  scope :for, ->(location) do
-    where(location: location)
+  def self.for(location)
+    @cache ||= {}
+
+    @cache[location] ||= where(location: location)
```
Comment (eanders) [inline app/models/link.rb:11]: These do change (there's an interface for adding menu items)  Not 100% sure it's the same items, I'm not remembering the `:for` scope, butt we should make sure to invalidate this cache if the items are added/removed.

File: app/controllers/admin/links_controller.rb [hunk 2]
```diff
@@ -28,11 +29,13 @@ def edit
 
     def update
       @link.update(link_params)
+      link_scope.invalidate_cache
       respond_with(@link, location: admin_links_path)
     end
 
     def destroy
       @link.destroy
+      link_scope.invalidate_cache
```
Comment (eanders) [inline app/controllers/admin/links_controller.rb:38]: @yakloinsteak I _think_ this handles the invalidation, look good to you?

Comment (yakloinsteak) [reply app/controllers/admin/links_controller.rb:38]: That looks great. I'm glad we could do it without redis in this particular case.

**Pull request merged**